fd/diff.go

```diff
@@ -19,25 +18,30 @@ type Point struct {
 	Coeff float64
 }
 
-// Method represents a finite difference formula that approximates
-// the derivative of order k of a function f at x as
-//  d^k f(x) ≅ (1 / h^k) * \sum_i Coeff_i * f(x + h * Loc_i),
-// where h is a small positive step.
-type Method struct {
+// Formula represents a finite difference formula on a regularly spaced grid
+// that approximates the derivative of order k of a function f at x as
+//  d^k f(x) ≅ (1 / Step^k) * \sum_i Coeff_i * f(x + Step * Loc_i).
+type Formula struct {
 	// Stencil is the set of sampling Points which are used to estimate the
 	// derivative. The locations will be scaled by Step and are relative to x.
 	Stencil []Point
 	Order   int     // The order of the approximated derivative.
-	Step    float64 // Default step size for the method.
+	Step    float64 // Default step size for the formula.
 }
 
 // Settings is the settings structure for computing finite differences.
 type Settings struct {
-	OriginKnown bool    // Flag that the value at the origin x is known
-	OriginValue float64 // Value at the origin (only used if OriginKnown is true)
-	Concurrent  bool    // Should the function calls be executed concurrently
-	Workers     int     // Maximum number of concurrent executions when evaluating concurrently
-	Method      Method  // Finite difference method to use
+	// Formula is the finite difference formula used
+	// for approximating the derivative.
+	Formula Formula
+	// Step is the distance between points of the stencil.
+	// If equal to 0, formula's default step will be used.
+	Step float64
```

Comment:
If `Step` is not in `Settings`, then using a custom step is too cumbersome: assign Formula to a variable, change the step, assign the formula to settings, call Derivative/Gradient. The usage example in `example_test.go` is quite nice I think.